src/core/opamSystem.ml

```diff
@@ -63,7 +86,7 @@ let rec mk_temp_dir ?(prefix="opam") () =
   if Sys.file_exists s then
     mk_temp_dir ()
   else
-    s
+    real_path s
```

Comment:
I think the only reason we have a wrapper around `OpamCompat.Unix.realpath` is because the OCaml < 4.13 version of it doesn't support files (as opposed to directories). In this case we know it is always a directory so it's safe to use that function directly.
```suggestion
    OpamCompat.Unix.realpath s
```